packages/blocks/src/_common/inline/presets/nodes/reference-node/reference-config.ts

```diff
@@ -0,0 +1,32 @@
+import type { Page } from '@blocksuite/store';
+import type { TemplateResult } from 'lit';
+
+export class ReferenceNodeConfig {
+  private _customIcon: TemplateResult | null = null;
+  private _customTitle: string | null = null;
+  private _page: Page | null = null;
+
+  get customIcon() {
+    return this._customIcon;
+  }
+
+  get customTitle() {
+    return this._customTitle;
+  }
+
+  get page() {
+    return this._page;
+  }
+
+  setCustomIcon(icon: TemplateResult | null) {
```

Comment:
The interface here only allows a single value for all reference-node rendered by a paragraph service, right?
From affine side what we may want here is to pass in a render function (possibly with delta + block context), instead of a single value.

Reply:
Do you need to retain the basic interactive logic, such as hover and click?

Reply:
Yes please.

Reply:
done

Reply:
Can you please declare the icon/title as a renderer function like

```ts
  setCustomContent(content: TemplateResult | ((context: { delta: Delta, block: Block } ) => TemplateResult)  | null) {
    this._customContent = content;
  }
```

It is not so useful if you don't give the context for downstream to know what to render

Reply:
done